src/Serilog/LoggerConfiguration.cs

```diff
@@ -153,35 +157,29 @@ public Logger CreateLogger()
             _maximumDestructuringDepth,
             _maximumStringLength,
             _maximumCollectionCount,
-            _additionalScalarTypes,
-            _additionalDictionaryTypes,
-            _additionalDestructuringPolicies,
+            _additionalScalarTypes ?? [],
+            _additionalDictionaryTypes ?? [],
+            _additionalDestructuringPolicies ?? [],
             auditing);
         var processor = new MessageTemplateProcessor(converter);
 
-        var enricher = _enrichers.Count switch
+        var enricher = _enrichers?.Count switch
         {
             // Should be a rare case, so no problem making that extra interface dispatch.
+            null => new EmptyEnricher(),
             0 => new EmptyEnricher(),
             1 => _enrichers[0],
             // Enrichment failures are not considered blocking for auditing purposes.
             _ => new SafeAggregateEnricher(_enrichers)
         };
 
         LevelOverrideMap? overrideMap = null;
-        if (_overrides.Count != 0)
+        if (_overrides is not null && _overrides.Count != 0)
         {
             overrideMap = new(_overrides, _minimumLevel, _levelSwitch);
         }
 
-        var disposableSinks = _logEventSinks
-            .Concat(_auditSinks)
-            .Where(s => s is IDisposable
-#if FEATURE_ASYNCDISPOSABLE
-                or IAsyncDisposable
-#endif
-                )
-            .ToArray();
+        var disposableSinks = Util.WhereIsDisposableSinks(_logEventSinks, _auditSinks).ToArray();
```

Comment:
I think this could be a nice change but would prefer to consider it separately and see whether any other code can benefit from an `IsDisposable` helper somewhere (which would then help organize the code).